tempesta_fw/sock.c

```diff
@@ -751,7 +755,7 @@ ss_tcp_process_skb(struct sock *sk, struct sk_buff *skb, int *processed)
 			continue;
 		}
 
-		r = SS_CALL(connection_recv, conn, skb, offset);
+		r = SS_CALL(connection_recv, conn, skb, off);
```

Comment:
Is there the need for an extra `off` variable, and extra assignments? Can we just do `offset = 0` after this line?

Reply:
In case of `Conn_Stop` flag set (https://github.com/tempesta-tech/tempesta/blob/8a40fda1111ad7e4ab7c6a2740e482e48e2f8986/tempesta_fw/sock.c#L731) - we will go to the next iteration, `offset` will remain in incorrect state. On the other hand, we must check `Conn_Stop` flag for connection - before `SS_CALL()`.

Reply:
Yep, got it. Thanks!

Reply:
On the other hand, once we have that condition, doesn't it mean that we never get to proceed past it (to `SS_CALL`), so `offset` will not be used anyway, and its value won't matter? We will just loop until we account for and free all skb in `skb_list`, and never execute `SS_CALL()`.

Reply:
On the other hand, once we have that condition, wouldn't it mean that we never get to proceed past it (to `SS_CALL()`), so that `offset` would not be used anyway, and its value would be irrelevant? We would just loop until we account for and free all skb in `skb_list`, and we would never execute `SS_CALL()`. In the end, wouldn't that mean, that we could just zero `offset` after it's first use (where you do it already) without the need for an extra `off` variable?

Reply:
Sorry for (possible) latest comments, my browser got stuck in reload/repost mode. I have nothing to add to the discussion.